### ISSUE TITLE
MagickImageCollection hangs when saving pdf

### ISSUE THREAD

**Issue by ssshuler** [dlemstra/Magick.NET#456]
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am using the latest version of Magick.NET

### System Configuration

- Magick.NET version: Magick.NET-Q8-AnyCPU v7.13.1
- Environment (Operating system, version and so on): Windows 7 Home Premium, Microsoft Visual Studio Community 2019, GhostScript 9.27

### Question

After recent system reinstall the code previously working fine hangs up on creating pdf file. The file gets created however with 0 length. I've tried to experiment with different configurations in regard to Q8/Q16 or x86/x64 settings and I've also installed the lastest Ghostscript both 32 and 64-bit version copying the exe/dll to my output folder all without much success. My test code goes as follows:

    public void SavePdfTest() {
         MagickNET.SetLogEvents(LogEvents.All);
         MagickNET.Log += MagickNET_Log;
         string path = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
         MagickNET.SetGhostscriptDirectory(path);
         string file1 = ResourceExtractor.GetResourceFile("640x480.jpg");
         string file2 = ResourceExtractor.GetResourceFile("480x600.jpg");
         Assert.IsTrue(File.Exists(file1));
         Assert.IsTrue(File.Exists(file2));
         string pdfFileName = Path.Combine(TestContext.TestDir, "test.pdf");
         using(var images = new MagickImageCollection()){
            images.Add(new MagickImage(file1));
            images.Add(new MagickImage(file2));
            try {
               images.Write(pdfFileName);
               Assert.IsTrue(File.Exists(pdfFileName));
            } catch (Exception exception) {
               Assert.Fail("Exception: {0}", exception.Message);
            }
         }

and it hangs on the line with images.Write() call. Debug output refers to Exception with reading some magick_opencl_....bin file from temp folder as attached
[imagemagick_debug.log](https://github.com/dlemstra/Magick.NET/files/3249083/imagemagick_debug.log)

This magick_opencl...bin does not exist, but there's another one with equally long name with short random sequence in the end.

How can I make it work again?

**Comment by dlemstra** [comment]
Are you sure that this is all that you are doing? Your log file contains lines that I would not expect with this code? Would it be possible to create a small project and publish that on GitHub so I can try to reproduce this?

**Comment by ssshuler** [comment]
Sure, please see https://github.com/ssshuler/ImageToPdfTest for standalone demo.

**Comment by dlemstra** [comment]
Thanks for the demo and I think I figured out why this is failing. It appears that when saving the PDF file ImageMagick will use the OpenCL library. When this is happening and you have upgraded the driver or got a new video card it will run a benchmark to determine if it is better to use the CPU or the GPU. Best advise that I can give you is to just keep the application running and wait till the benchmark is done.

**Comment by ssshuler** [comment]
Hmm. That does make sense as I also switched to built-in graphics card. This also explains this annoying problem I'm seeing with a message that Intel HD Graphics driver recovered from an error. The bottom line though is that no matter how long I wait this benchmarking never finishes, most likely because of the driver problem as above.

I've just checked and I have all updates installed. Is there any way to force the use of CPU in the code to circumvent the benchmark?
